### ISSUE TITLE
enableProposedApi not documented in package.json

### ISSUE THREAD

**Issue by isidorn** [Microsoft/vscode#33358]
Refs: #33248

`enableProposedApi` is not suggested in package.json inteli-sense
Due to that it is hard to discover this flag. Is this on purpose?

**Comment by jrieken** [comment]
Yeah, that is actually on purpose because we don't want to over-promote this. Still, we should document somewhere how proposed API to be used

**Comment by aeschli** [comment]
IMO we should add it. The description can explain the limitations you get when you use it. Given the limitations, there's IMO no danger that too many devs will use it.

**Comment by octref** [comment]
@jrieken Can we include it in a section under `Extensibility Reference`?
![image](https://user-images.githubusercontent.com/4033249/43611152-11047b46-965d-11e8-8ebb-0b430f746d76.png)

I don't want to go into GitHub comments to find out the instruction for using proposed API each time.

If that's fine, I can create a placeholder page in the website and you can fill in details.

**Comment by jrieken** [comment]
:+1: for the documentation and :-1: for the sample. That will be out of date in no-time and I also don't think that it adds value

**Comment by octref** [comment]
@jrieken However, currently it's awkward to update your code to use Proposed API. Each time I had to remember to do this:

- Find latest `vscode.d.ts`
- Change `vscode.engines` to `*`
- Run `postinstall` scripts to download latest d.ts
- Change `vscode.engines` back
- Remember to add `enabledProposedApi` to package.json. I sometimes type it wrongly as `API` and nothing works.

If sample is not the right place, what do you think about adding another script to the `vscode` NPM module to automate this? So people can run `node ./node_modules/vscode/use-proposed-api`. We can print warnings and have confirmations here too.

**Comment by jrieken** [comment]
@octref step 1 to 4 have nothing to do with proposed api. that's the procedure when coding against the latest, not yet published, version of `vscode.d.ts` and is only rarely needed. 

for proposed API the three steps outline here are needed: https://code.visualstudio.com/updates/v1_28#_proposed-extension-apis

**Comment by octref** [comment]
OK I'll drop the sample.

**Comment by mjbvz** [comment]
Should this issue be reopened or marked as wont-fix?

**Comment by octref** [comment]
To verify:

- Verify the content https://vscode-ext-docs.azurewebsites.net/api/advanced-topics/using-proposed-api makes sense to you.
- Verify the top comment in `vscode.proposed.d.ts` makes sense to you.
